src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/ExtensionRole.cs

```diff
@@ -22,12 +23,28 @@ public class ExtensionRole
         protected const string DefaultExtensionIdPrefixStr = "Default";
         protected const string ExtensionIdSuffixTemplate = "-{0}-{1}-Ext-{2}";
         protected const int MaxExtensionIdLength = 60;
+        protected const int MaxSuffixLength = MaxExtensionIdLength - 1;
 
         public string RoleName { get; private set; }
         public string PrefixName { get; private set; }
         public ExtensionRoleType RoleType { get; private set; }
         public bool Default { get; private set; }
 
+        private static string RemoveDisallowedCharacters(string roleName)
+        {
+            // Remove characters that are not allowed in the extension id
+            var disallowedCharactersRegex = new Regex(@"[^A-Za-z0-9\-]");
+            var match = disallowedCharactersRegex.Match(roleName);
+
+            while (match.Success)
+            {
+                roleName = roleName.Remove(match.Index, match.Length);
+                match = disallowedCharactersRegex.Match(roleName);
+            }
+
+            return roleName;
+        }
```

Comment:
this method seems too complex for the thing it is doing. 
Just put following on line 37 and you should be good:
```
return disallowedCharactersRegex.Replace(roleName,"");
```